### PR TITLE
Add docs for System.Transactions

### LINKED ISSUE #619: Update transactions documentation with System.Transaction support details
We can probably add a section to https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/saving/transactions.md. We should call out any limitations.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #619

### CODE REVIEW

File: samples/core/Saving/Saving/Transactions/AmbientTransaction/Sample.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                new DbContextOptionsBuilder<BloggingContext>()
+                    .UseSqlServer(connectionString)
+                    .Options))
+            {
+                context.Database.EnsureDeleted();
+                context.Database.EnsureCreated();
+            }
+
+            #region Transaction
+            using (var scope = new TransactionScope(TransactionScopeOption.Required))
```
Comment (divega) [inline samples/core/Saving/Saving/Transactions/AmbientTransaction/Sample.cs]: Mind if I change the isolation level? The default isn't great. See @davidbaxterbrowne's write up for reference: https://blogs.msdn.microsoft.com/dbrowne/2010/06/03/using-new-transactionscope-considered-harmful/

File: entity-framework/core/saving/transactions.md [hunk 2]
```diff
@@ -111,81 +101,25 @@ public class BloggingContext : DbContext
...
-                // Commit transaction if all commands succeed, transaction will auto-rollback
-                // when disposed if either commands fails
-                transaction.Commit();
-            }
-            catch (System.Exception)
-            {
-                // TODO: Handle failure
-            }
-```
+[!code-csharp[Main](../../../samples/core/Saving/Saving/Transactions/CommitableTransaction/Sample.cs?name=Transaction&highlight=1,13,26,27,28)]
```
Comment (divega) [inline entity-framework/core/saving/transactions.md]: I would like to add a note explaining that as of version 2.1, .NET Core does not support using System.Transactions to coordinate transactions distributed across multiple resource managers. Sounds good?

Comment (AndriySvyryd) [reply entity-framework/core/saving/transactions.md]: There are probably some other limitations, but I can't recall now.
For instance we could mention that non-relational providers might not support this.

Comment (divega) [reply entity-framework/core/saving/transactions.md]: Fair enough. We should said that it is up to the provider in general.

**Pull request merged**